doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1737,33 +1779,81 @@ Controlling the proof flow
    impossible. In most cases, :g:`P` or :g:`∼P` is one of the hypotheses of the
    local context.
 
-.. tacn:: contradiction
-   :name: contradiction
+.. tacn:: contradiction {? @one_term_with_bindings }
 
-   This tactic applies to any goal. The contradiction tactic attempts to
-   find in the current context (after all intros) a hypothesis that is
-   equivalent to an empty inductive type (e.g. :g:`False`), to the negation of
-   a singleton inductive type (e.g. :g:`True` or :g:`x=x`), or two contradictory
-   hypotheses :g:`P` and :g:`not P`.
+   Tries to prove the current goal by finding a contradiction.
 
-   .. exn:: No such assumption.
-      :undocumented:
+   If :n:`@one_term_with_bindings` is not provided (the most common use case),
+   the tactic first does an :tacn:`intros`.  The tactic then proves the goal if
+
+   - the updated context has a pair of hypotheses where one is the negation of
+     the other (e.g. :n:`P` and not :n:`~P`), or
+   - there is a hypothesis with an empty inductive type (e.g. :n:`False`), or
+   - there is a hypothesis :n:`~P` where `P` is a singleton inductive type
+     (e.g. :n:`True` or :n:`x=x`) provable by `Goal P. constructor.`.
+
+   If :n:`@one_term_with_bindings` is provided, its type
+   must be a negation, such as :n:`~P`,
+   or an empty inductive type, such as :n:`False`.
+   If the type is a negation and :n:`P` is a hypothesis in the context,
+   the goal is proven.  If the type is a negation and :n:`P` is not in
+   the context, the goal is replaced with :n:`P`.  If the type is :n:`False`
+   or another empty inductive type, the goal is proven.
+   Otherwise the tactic fails.  (If there is a hypothesis
+   `P` and you want to replace the goal with `~P`, use the :tacn:`contradict`
+   tactic.  If there are hypotheses `H1 : P` and `H2 : ~P`, use `contradiction`
+   without arguments or `contradiction H2`; `contradiction H1` won't work.)
```

Comment:
I suggest replacing ";" with a word, otherwise it looks like what is written is:

>or `contradiction H2; contradiction H1` won't work

![image](https://user-images.githubusercontent.com/1108325/209586937-f20156f4-ac46-4d34-a701-c000ed7fc301.png)

```suggestion
   without arguments or `contradiction H2` since `contradiction H1` won't work.)
```